kernel/tests/dv.rs

```diff
@@ -1,12 +1,30 @@
 //! Read a small table with/without deletion vectors.
 //! Must run at the root of the crate
+use std::ops::Add;
 use std::path::PathBuf;
 
 use delta_kernel::engine::sync::SyncEngine;
-use delta_kernel::Table;
+use delta_kernel::scan::ScanResult;
+use delta_kernel::{DeltaResult, Table};
 
+use itertools::Itertools;
 use test_log::test;
 
+fn count_valid_rows(stream: impl Iterator<Item = DeltaResult<ScanResult>>) -> DeltaResult<usize> {
```

Comment:
I wanted to flag this helper function I made, since I noticed duplicate code. This handles `res` being a `DeltaResult`, avoids `let res = res?`. Instead of having mutable state that a `map_ok` accesses, I made it into a sum over each RecordBatch.